default.nix

```diff
@@ -0,0 +1,106 @@
+{ lib, fetchFromGitHub, stdenv, cmark, file, gettext, git,
+  makeWrapper, runCommand, python27, re2c, readline }:
+let
+	py-yajl = python27.pkgs.buildPythonPackage rec {
+		pname = "oil-pyyajl";
+		version = "unreleased";
+		src = fetchFromGitHub {
+			owner = "oilshell";
+			repo = "py-yajl";
+			rev = "eb561e9aea6e88095d66abcc3990f2ee1f5339df";
+			sha256 = "17hcgb7r7cy8r1pwbdh8di0nvykdswlqj73c85k6z8m0filj3hbh";
+			fetchSubmodules = true;
+		};
+		nativeBuildInputs = [ git ];
+	};
+
+	oilPython = python27.withPackages (ps: with ps; [ six typing ]);
+
+	oildev = python27.pkgs.buildPythonPackage rec {
+		pname = "oil";
+		version = "undefined";
+
+		# src = builtins.fetchGit ../../../../work/oil;
+		src = fetchFromGitHub {
+			owner = "abathur";
+			repo = "oil";
+			rev = "259e582598689cb5077c44819f3234dda79c34fa";
+			sha256 = "0rx68y8r82sr8qmbr806iaz2pispn02f64k6xywxpj5lx05jynlz";
+		};
+
+		# src = fetchFromGitHub {
+    #   owner = "oilshell";
+		#   repo = "oil";
+		#   rev = "58f2372abd7df45221c0b74239cdc4442dbb8906";
+		#   sha256 = "15qmkhkj7cc1kb0c42vshddmd484yi5x2xh4826i033drf3iqryw";
+		#   fetchSubmodules = true;
+		# };
+
+		buildInputs = [ oilPython readline re2c cmark py-yajl makeWrapper ];
+		# buildInputs = [ ];
+		nativeBuildInputs = [ re2c file oilPython py-yajl ];
+
+		# runtime deps
+		propagatedBuildInputs = [ re2c oilPython py-yajl ];
+
+		doCheck = true;
+		dontStrip = true;
+
+		preBuild = ''
+			echo $buildPhase
+			set -x
+			build/dev.sh all
+			set +x
+		'';
+
+		# Patch shebangs so Nix can find all executables
+		postPatch = ''
+	      patchShebangs .
```

Comment:
Oops. Looks like I mixed up tabs and spaces here.

Reply:
no worries; going to pick at it anyways